combat/CombatSystem.cpp

```diff
@@ -1266,12 +1310,38 @@ namespace {
 
 
             // select target object
-            int target_idx = RandInt(0, valid_target_ids.size() - 1);
-            int target_id = *std::next(valid_target_ids.begin(), target_idx);
+            std::shared_ptr<UniverseObject> target;
+            DebugLogger() << "Target precision: " << std::to_string(weapon.precision);
+            for (int t = 1; t < weapon.precision ; t++) {
+                // Do all but the last round of finding a valid target; the simple last round is separate for performance reasons
+                DebugLogger() << "Find target - Round: " << std::to_string(t) << " (try to lock it if it matches preferredPrey conditions)";
+                int target_idx = RandInt(0, valid_target_ids.size() - 1);
+                int target_id = *std::next(valid_target_ids.begin(), target_idx);
```

Comment:
is this OK if `valid_target_ids` has just one element?

Reply:
Well that was already the old code. And playtesting it worked if there was only one enemy. So I'd say its OK ;)